Demo/MozillaSocial-iOS/MoSoContent/Auth.swift

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AuthenticationServices
+
+let isProd = false
+let baseURL = isProd ? "https://mozilla.social" : "https://stage.moztodon.nonprod.webservices.mozgcp.net"
+let redirectScheme = "mozillasocial"
+let redirectURI = redirectScheme + "://auth"
+
+class Auth: NSObject, ASWebAuthenticationPresentationContextProviding {
+
+    var authToken: Token?
+
+    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        return ASPresentationAnchor()
+    }
+
+    func registerApp() async -> ClientEntity? {
+        var urlRequest = URLRequest(url: registerAppURL())
```

Comment:
Ideally, we could think of expanding `MoSoClient` and let it take care of network requests, but given that the entire auth process is still under construction, I guess we can leave it self contained here (asking @dskuza for his thoughts here)

Reply:
This authentication code will be superseded in the near future. Keeping this code as contained as possible will make life easier when that time comes.